lib/services/local/users.go

```diff
@@ -1388,9 +1389,71 @@ func (s *IdentityService) GetMFADevices(ctx context.Context, user string, withSe
 		}
 		devices = append(devices, &d)
 	}
+
```

Comment:
@tigrato I just happened upon your [PR](https://github.com/gravitational/teleport/pull/46957) adding weakest MFA device logic. Should we incorporate SSO MFA into this weakest device logic?

In general, SSO MFA should be considered stronger than TOTP but weaker than Webauthn, but really it depends on the implementation. If the admin sets it up properly, they'll have WebAuthn or stronger/equivalent (custom hardware keys like bloomber's b-units. Worst case it uses their SSO login session to re-auth them, or prompts them to re-login via SSO.

The main difficulty is that we don't actually create these MFA devices for the user. When MFA is enabled for the user's auth connector, it makes it so `GetMFADevices` includes an ephemeral SSO MFA device (shown here). So whenever an auth connector is updated with `mfa.enabled = true/false`, we would need to loop through all users created by the connector and call `upsertUserStatusMFADevice(ctx, user)` to update their weakest MFA device state. WDYT?

Reply:
@tigrato bump on this concern, will things break as is - without adding SSO MFA to the weakest device logic?